chalice/app.py

```diff
@@ -264,17 +273,40 @@ def __init__(self, body, headers=None, status_code=200):
         if headers is None:
             headers = {}
         self.headers = headers
+        self.case_insensitive_headers = CaseInsensitiveMapping(headers)
```

Comment:
Why are we adding a case insensitive headers?  Seems confusing to have both, and it's not obvious when you'd use one or the other.  I suggest we have one attr, `headers`, and that's it.